scripts/flaky_tests.py

```diff
@@ -0,0 +1,90 @@
+#! /usr/bin/env python3
+
+import argparse
+import json
+import logging
+from collections import defaultdict
+from typing import DefaultDict
+
+import psycopg2
+import psycopg2.extras
+
+# We call the test "flaky" if it changed its status on the same revision in the last N=10 days on main branch.
```

Comment:
Hmm, I always treated the flakiness of tests per run. Interesting

Reply:
Do you suggest rerunning any failed tests to check if it's flaky or not? I don't really like this approach because it hides real failures (like race conditions, for example). 
We could probably try something like this along with https://github.com/neondatabase/neon/pull/3907, but I'd like to start with something less permissive. 

For now, we can simplify the query to find tests that failed at least once in the last ten days on the main branch. Since we don't allow merging branches with failed tests, this should be enough (a bit less accurate than with revision, but also doesn't require human interaction for marking tests as flaky).
Let's use this one, gather feedback and adjust the logic accordingly.

Reply:
I'm ok to start with your approach and measure its success. I don't have any prejudices :)